web-app-router/scenes/Portal/Teams/TeamId/Team/page/Members/List/EditRoleDialog/index.tsx

```diff
@@ -0,0 +1,244 @@
+"use client";
+
+import { CircleIconContainer } from "@/components/CircleIconContainer";
+import { Dialog } from "@/components/Dialog";
+import { DialogOverlay } from "@/components/DialogOverlay";
+import { DialogPanel } from "@/components/DialogPanel";
+import { UserEditIcon } from "@/components/Icons/UserEditIcon";
+import { atom, useAtom } from "jotai";
+import { Notification } from "@/components/Notification";
+import { TYPOGRAPHY, Typography } from "@/components/Typography";
+import { Link } from "@/components/Link";
+import { Controller, useForm } from "react-hook-form";
+import { Input } from "@/components/Input";
+import * as yup from "yup";
+import { Role_Enum } from "@/graphql/graphql";
+import { yupResolver } from "@hookform/resolvers/yup";
+import {
+  Select,
+  SelectButton,
+  SelectOption,
+  SelectOptions,
+} from "@/components/Select";
+import { DecoratedButton } from "@/components/DecoratedButton";
+import {
+  FetchMembershipsDocument,
+  FetchMembershipsQuery,
+} from "../graphql/client/fetch-members.generated";
+import { useCallback, useEffect, useMemo } from "react";
+import { CaretIcon } from "@/components/Icons/CaretIcon";
+import { useEditRoleMutation } from "./graphql/client/edit-role.generated";
+import { useParams } from "next/navigation";
+import { toast } from "react-toastify";
+import { Button } from "@/components/Button";
+import { permissionsDialogAtom } from "../PermissionsDialog";
+import clsx from "clsx";
+
+export const editRoleDialogAtom = atom(false);
+
+const schema = yup.object({
+  email: yup.string().nullable(),
+  role: yup.object({
+    label: yup.string(),
+    value: yup.string().oneOf(Object.values(Role_Enum)),
+  }),
+});
+
+type FormValues = yup.InferType<typeof schema>;
+
+export const EditRoleDialog = (props: {
+  membership: FetchMembershipsQuery["membership"][number] | null;
+}) => {
+  const [isOpened, setIsOpened] = useAtom(editRoleDialogAtom);
+  const [permissionsOpened, setPermissionsOpened] = useAtom(
+    permissionsDialogAtom,
+  );
+  const { teamId } = useParams() as { teamId: string };
+
+  const roles = useMemo(
+    () =>
+      Object.entries(Role_Enum).map(([key, value]) => ({
+        label: key,
+        value,
+      })),
+    [],
+  );
+
+  const defaultValues = useMemo(() => {
+    if (!props.membership) {
+      return null;
+    }
+
+    return {
+      email: props.membership?.user.email,
+      role: roles.find((role) => role.value === props.membership?.role),
+    };
+  }, [props.membership, roles]);
+
+  const {
+    register,
+    control,
+    formState: { isDirty, isSubmitting },
+    handleSubmit,
+    reset,
+  } = useForm({
+    resolver: yupResolver(schema),
+    mode: "onChange",
+    defaultValues: {
+      email: "",
+      role: { label: "", value: Role_Enum.Member },
+    },
+  });
+
+  useEffect(() => {
+    if (defaultValues) {
+      reset(defaultValues);
+    }
+  }, [defaultValues, props.membership, reset]);
```

Comment:
```suggestion
  }, [defaultValues reset]);
```
we dont need props.membership here right since default value is dependent on it already